app/modules/code_provider/github/github_service.py

```
@@ -506,22 +506,51 @@
+                    branch_list = [
+                        branch.name for branch in branches if branch.name != default_branch
+                    ]
+                    return {"branches": [default_branch] + branch_list}
+                except Exception as e:
+                    # Check if the error is a GitHub API exception or not
+                    if hasattr(e, "status"):
+                        if e.status == 403:
+                            raise HTTPException(
+                                status_code=403,
+                                detail=f"Access denied to repository {repo_name}. You don't have sufficient permissions to access this repository."
+                            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)